### PR TITLE
feat(ReactComponentLibrary): Expose new stable implementations

### LINKED ISSUE #3043: Update composite components to use new implementations ahead of release

### LINKED ISSUE #3044: Remove deprecated components and update experimental component namespaces

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #3044
Closes #3043

## Overview

Expose experimental implementations as stable.

## Link to preview

https://620f8cfadd0ed914212ef643--festive-wozniak-29a757.netlify.app

## Reason

>Prime code base for 3.x.x release. Replacing deprecated implementations with new stable ones and updating composite components to consume the new implementations.

## Work carried out

- [x] Remove deprecated components and update experimental namespace
- [x] Update composite components to use new implementations
- [x] Fix display bug with Drawer Story

### CODE REVIEW

File: packages/react-component-library/cypress/selectors/index.ts [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
 import contextMenu from './ContextMenu'
 import datePicker from './DatePicker'
 import dismissibleBanner from './DismissibleBanner'
 import form from './form'
-import numberInputE from './NumberInputE'
-import rangeSliderE from './RangeSliderE'
-import selectE from './SelectE'
+import NumberInput from './NumberInput'
+import RangeSlider from './RangeSlider'
+import selectE from './Select'
```
Comment (jpveooys) [inline packages/react-component-library/cypress/selectors/index.ts]: Still a `selectE` here:

```suggestion
import select from './Select'
```

File: packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx [hunk 2]
```diff
@@ -1,16 +1,14 @@
 import React from 'react'
 
 import { ButtonGroupItemProps } from '.'
-import { BUTTON_SIZE } from '../Button'
+import { COMPONENT_SIZE } from '../Forms'
 import { ComponentWithClass } from '../../common/ComponentWithClass'
 import logger from '../../utils/logger'
 import { StyledButtonGroup } from './partials/StyledButtonGroup'
 
 export type ButtonGroupSizeType =
```
Comment (jpveooys) [inline packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx]: Are we able to get rid of this and use `ComponentSizeType` instead?

**Review by jpveooys — COMMENTED** [review]
Nice work!

Will have another look through but these are a couple of minor things I spotted.

File: packages/react-component-library/cypress/selectors/index.ts [hunk 3]
```diff
@@ -1,25 +1,25 @@
-import autocompleteE from './AutocompleteE'
+import Autocomplete from './Autocomplete'
 import avatar from './Avatar'
-import checkboxE from './CheckboxE'
+import Checkbox from './Checkbox'
```
Comment (jpveooys) [inline packages/react-component-library/cypress/selectors/index.ts]: Would be good to keep the casing consistent here (e.g. `avatar` vs `Checkbox`)

File: packages/react-component-library/cypress/specs/Select/index.spec.ts [hunk 4]
```diff
@@ -2,40 +2,40 @@ import { describe, cy, it, before } from 'local-cypress'
...
+        cy.get(selectors.Select.outerWrapper).type('Th{enter}')
         cy.get('body').click()
       })
 
       it('sets the value as the item', () => {
-        cy.get(selectors.selectE.input).should(
+        cy.get(selectors.Select.input).should(
           'have.value',
-          'This is a really, really long select option label that overflows the container when selected'
+          'This is a really, really long select option label that overflows the container when Selectd'
```
Comment (jpveooys) [inline packages/react-component-library/cypress/specs/Select/index.spec.ts]: Looks like this got caught up in the changes :smile: 
```suggestion
          'This is a really, really long select option label that overflows the container when selected'
```

**Review by jpveooys — COMMENTED** [review]
Looks good to me overall! There are a few build failures at the moment though.

**Comment by jpveooys** [comment]
Looks like we can get rid of `react-responsive` and `@types/react-responsive` now (was being used by `Switch`).

**Review by jpveooys — APPROVED** [review]

**Comment by k9dh3zij** [comment]
Looking great! 🥳

Tiny detail: https://620f8cfadd0ed914212ef643--festive-wozniak-29a757.netlify.app/?path=/docs/masthead--default
Can we change "Defence Digital Design System" to "MOD.UK Design System" on all stories please? 

Also: wondering if we need both `Panel` and `Card frame`? I think we discussed getting rid of one of them, but that was a while back.

Thanks!

**Pull request merged**